feature-libs/quote/components/actions/by-role/quote-actions-by-role.component.ts

```diff
@@ -0,0 +1,235 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  Component,
+  ElementRef,
+  inject,
+  OnDestroy,
+  OnInit,
+  ViewChild,
+  ViewContainerRef,
+} from '@angular/core';
+import { ActiveCartFacade, Cart } from '@spartacus/cart/base/root';
+import { GlobalMessageService, GlobalMessageType } from '@spartacus/core';
+import {
+  Quote,
+  QuoteAction,
+  QuoteActionType,
+  QuoteFacade,
+  QuoteRoleType,
+  QuoteState,
+} from '@spartacus/quote/root';
+import { LAUNCH_CALLER, LaunchDialogService } from '@spartacus/storefront';
+import { Observable, Subscription } from 'rxjs';
+import { filter, take, tap } from 'rxjs/operators';
+import {
+  ConfirmActionDialogConfig,
+  QuoteUIConfig,
+} from '../../config/quote-ui.config';
+import { ConfirmationContext } from '../confirm-dialog/quote-actions-confirm-dialog.model';
+
+@Component({
+  selector: 'cx-quote-actions-by-role',
+  templateUrl: './quote-actions-by-role.component.html',
+})
+export class QuoteActionsByRoleComponent implements OnInit, OnDestroy {
+  protected quoteFacade = inject(QuoteFacade);
+  protected launchDialogService = inject(LaunchDialogService);
+  protected viewContainerRef = inject(ViewContainerRef);
+  protected globalMessageService = inject(GlobalMessageService);
+  protected quoteUIConfig = inject(QuoteUIConfig);
+  protected activeCartFacade = inject(ActiveCartFacade);
+
+  quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
+  cartDetails$: Observable<Cart> = this.activeCartFacade.getActive();
+
+  @ViewChild('element') element: ElementRef;
+  QuoteActionType = QuoteActionType;
+  protected subscription = new Subscription();
+
+  ngOnInit(): void {
+    //submit button present and threshold not reached: Display message
+    this.quoteDetails$.pipe(take(1)).subscribe((quote) => {
+      const mustDisableAction = quote.allowedActions.find((action) =>
+        this.mustDisableAction(action.type, quote)
+      );
+      if (mustDisableAction) {
+        this.globalMessageService.add(
+          {
+            key: 'quote.commons.minRequestInitiationNote',
+            params: {
+              minValue: quote.threshold,
+            },
+          },
+          GlobalMessageType.MSG_TYPE_WARNING
+        );
+      }
+    });
+  }
+
+  mustDisableAction(type: string, quote: Quote): boolean {
+    return type === QuoteActionType.SUBMIT && !this.isThresholdReached(quote);
+  }
+
+  protected isThresholdReached(quote: Quote): boolean {
+    const requestThreshold = quote.threshold || 0;
+    return (quote.totalPrice.value || 0) >= requestThreshold;
+  }
+
+  onClick(action: QuoteActionType, quote: Quote, cart: Cart) {
```

Comment:
Please, add a comment to the method with an explanation what happens when the button is clicked.

Reply:
@Uli-Tiger, could you have a look?